src/components/PoolingWidget/index.tsx

```diff
@@ -207,24 +210,38 @@ const PoolingInterface: React.FC = () => {
 
   const { isSubmitting, placeMultipleOrders } = usePlaceOrder()
 
-  const onSentTransaction = useCallback(
-    txHash => {
-      setTxHash(txHash)
-      // nextStep()
-    },
-    [setTxHash],
-  )
-
   const sendTransaction = useCallback(async () => {
+    if (!networkId || !userAddress) return
     const orders = createOrderParams(Array.from(selectedTokensMap.values()), spread)
-
+    let pendingTxHash: string | undefined
     try {
       setTxReceipt(undefined)
 
       const { receipt } = await placeMultipleOrders({
         orders,
         txOptionalParams: {
-          onSentTransaction,
+          onSentTransaction: (txHash: string): void => {
+            pendingTxHash = txHash
+            setTxHash(txHash)
+
+            orders.map(({ buyToken: buyTokenId, sellToken: sellTokenId, buyAmount, sellAmount }) => {
+              const newTxState = {
+                txHash,
+                id: 'PENDING ORDER',
+                buyTokenId,
+                sellTokenId,
+                priceNumerator: buyAmount,
+                priceDenominator: sellAmount,
+                user: userAddress,
+                remainingAmount: ZERO,
+                sellTokenBalance: ZERO,
+                validFrom: 0,
+                validUntil: 0,
+              }
+
+              return dispatch(savePendingOrdersAction({ orders: newTxState, networkId, userAddress }))
```

Comment:
I wonder if multiple dispatches in a row cause rerenders
If you could check and maybe try with `unstable_batchedUpdates`

Reply:
:+1: